tests/RequestHeaderParserTest.php

```diff
@@ -113,6 +113,27 @@ public function testHeaderOverflowShouldEmitError()
         $this->assertSame('Maximum header size of 4096 exceeded.', $error->getMessage());
     }
 
+    public function testGuzzleRequestParseException()
+    {
+        $error = null;
+
+        $parser = new RequestHeaderParser();
+        $parser->on('headers', $this->expectCallableNever());
+        $parser->on('error', function ($message) use (&$error) {
+            $error = $message;
+        });
+
+        $this->assertSame(1, count($parser->listeners('headers')));
+        $this->assertSame(1, count($parser->listeners('error')));
+
+        $parser->feed("\r\n\r\n");
+
+        $this->assertInstanceOf('InvalidArgumentException', $error);
+        $this->assertSame('Invalid message', $error->getMessage());
+        $this->assertSame(0, count($parser->listeners('headers')));
+        $this->assertSame(0, count($parser->listeners('error')));
```

Comment:
I think my previous comment still holds: If we check the listeners for this error, we should probably also check the listeners for the other `error` event, so this provides a consistent interface for *any kind of error*.